contents/blog/clickhouse-vs-elasticsearch.md

```diff
@@ -0,0 +1,230 @@
+featuredImageType: full
+category: Engineering
+tags:
+  - Explainers
+  - ClickHouse
+---
+
+Elasticsearch and ClickHouse are both open-source frameworks with advantages over conventional databases like [PostgreSQL](/blog/clickhouse-vs-postgres) for performing tasks over *lots* of data, but they serve very different needs.
+
+Elasticsearch is the bedrock product for Elastic. It sells Elastic Cloud, a managed solution that bundles Elasticsearch with other data products. 
+
+![Elasticsearch Development.png](../images/blog/clickhouse-vs-elastic/elasticsearch-development.png)
+
+ClickHouse isn’t too different. While the database, ClickHouse, remains open source, it is managed by the for-profit ClickHouse Inc. ClickHouse Inc.’s main offering is ClickHouse Cloud, a managed service similar to Elastic Cloud, just for deploying ClickHouse instead. However, ClickHouse also merges notable contributions by Altinity, a separate company that sells Altinity.Cloud, a managed service for deploying ClickHouse in Kubernetes. 
+
+![ClickHouse Development.png](../images/blog/clickhouse-vs-elastic/clickhouse-development.png)
+
+Elasticsearch, as the name implies, was designed to power better search. It can efficiently return search results, such as grocery items on a grocer’s website, accounting for things such as spelling mistakes. ClickHouse, meanwhile, would stumble with the same task, but it excels at aggregating data for uses like business analytics or financial statistics. 
+
+Elasticsearch and ClickHouse are interesting to compare because of their vastly different architecture, optimized for each of their respective goals. Comparing them is a good meditation on physical and virtual layouts can improve efficiency toward a specific efficiency goal. 
+
+## Background
+
+Sometimes, the relationship between an open-source tool and its lead developer is complicated. ClickHouse's relationship is straightforward, but Elastic has a complex history with open source. 
+
+### What is Elasticsearch?
+
+Elasticsearch was originally released in 2010 under an open-source license. The premise behind Elasticsearch was that Apache Lucene, an open-source product designed to efficiently search JSON documents, needed better infrastructure for scaling. Apache Lucene made it easy to organize and search a series of JSON documents – such as human profiles; Elasticsearch made it easy to distribute those human profiles, which might be in the billions, across multiple locations, indexed both physically and virtually.
+
+Elasticsearch is considered a NoSQL database because it uses Apache Lucene – and by extension, JSON documents – as a primary store of data. Specifically, it is a Document-Store NoSQL database with a focus on searching and retrieving data. It is never used as the primary store of data. Sometimes Elasticsearch data stores are redundantly available in a more traditional database like PostgreSQL as Elasticsearch is only leveraged to improve search results. 
```

Comment:
```suggestion
Elasticsearch is considered a NoSQL database because it uses Apache Lucene – and by extension, JSON documents – as a primary store of data. Specifically, it is a Document-Store NoSQL database with a focus on searching and retrieving data. It is never used as the primary store of data. Elasticsearch data stores are often redundantly available in a more traditional database like PostgreSQL as Elasticsearch is only leveraged to improve search results. 
```